src/v0/destinations/klaviyo/util.js

```diff
@@ -123,33 +119,27 @@ const addUserToList = (message, traitsInfo, conf, destination) => {
 };
 
 /**
- * This function is used to check if the user needs to be added to list or needs to be subscribed or not.
- * Building and returning response array for both the members(for adding to the list) and subscribe
- * endpoints (for subscribing)
+ * This function is used to check if the user needs to be subscribed or not.
+ * Building and returning response array for subscribe endpoint (for subscribing)
  * @param {*} message
  * @param {*} traitsInfo
  * @param {*} destination
  * @returns
  */
-const checkForMembersAndSubscribe = (message, traitsInfo, destination) => {
+const checkForSubscribe = (message, traitsInfo, destination) => {
   const responseArray = [];
   if (
-    (!!destination.Config.listId || !!get(traitsInfo.properties, 'listId')) &&
-    destination.Config.privateApiKey
+    get(traitsInfo.properties, 'subscribe') === true &&
```

Comment:
Can we do this instead of using `get` function ?
```suggestion
traitsInfo?.properties?.subscribe && 
```

Reply:
Updated